src/main/flight/pid.c

```diff
@@ -445,19 +446,63 @@ STATIC_UNIT_TESTED FAST_CODE_NOINLINE float pidLevel(int axis, const pidProfile_
 #ifdef USE_GPS_RESCUE
     angle += gpsRescueAngle[axis] / 100; // ANGLE IS IN CENTIDEGREES
 #endif
-    angle = constrainf(angle, -levelAngleLimit, levelAngleLimit);
     const float errorAngle = angle - ((attitude.raw[axis] - angleTrim->raw[axis]) / 10.0f);
+    float angleErrorCorrection = errorAngle * pidRuntime.levelGain;
```

Comment:
Please refactor this so we can log the attitude of the quad using the input to errorAngle.  ie change to:
```
    const float currentAngle = (attitude.raw[axis] - angleTrim->raw[axis]) / 10.0f;
    const float errorAngle = angle - currentAngle;
```
we can then use a debug to log `currentAngle`, which gives us the true 'target' angle of the quad, and is more accurate than using `heading` as the comparison in the log.

Reply:
refactored as you suggest and added currentAngle debug.